platform_disk_api/admission_controller/api.py

```diff
@@ -1,43 +1,316 @@
+from __future__ import annotations
+
+import base64
+import json
 import logging
+from collections.abc import AsyncIterator
+from collections.abc import Awaitable
+from collections.abc import Callable
+from contextlib import AsyncExitStack
+from dataclasses import dataclass, field
+from enum import StrEnum
+from typing import Any
 
-import aiohttp
-import aiohttp.web
+from aiohttp import web
+from aiohttp.web_exceptions import HTTPConflict, HTTPBadRequest
+from apolo_kube_client.errors import ResourceExists
 
+from ..api import create_kube_client
 from ..config import Config
+from ..kube_client import KubeClient, DiskNaming
+from ..service import (
+    DISK_API_MARK_LABEL,
+    APOLO_DISK_API_MARK_LABEL,
+    APOLO_DISK_API_NAME_ANNOTATION,
+    DISK_API_NAME_ANNOTATION,
+    Service,
+    DISK_API_ORG_LABEL,
+    APOLO_ORG_LABEL,
+    DISK_API_PROJECT_LABEL,
+    APOLO_PROJECT_LABEL,
+    APOLO_DISK_API_CREATED_AT_ANNOTATION,
+    DISK_API_CREATED_AT_ANNOTATION,
+    USER_LABEL,
+    APOLO_USER_LABEL,
+)
+from ..utils import datetime_dump, utc_now
 
 LOGGER = logging.getLogger(__name__)
 
+PATH_ANNOTATIONS = "/metadata/annotations"
+PATH_LABELS = "/metadata/labels"
+
+KUBE_CLIENT_KEY = web.AppKey("kube_client", KubeClient)
+CONFIG_KEY = web.AppKey("config", Config)
+
+
+def escape_json_pointer(path: str) -> str:
+    """
```

Comment:
I think this function is already defined in apolo_kube_client

Reply:
@zubenkoivan apolo-kube-client is not yet bumped to the newest version. I propose to this as a follow-up because it'll require refactoring of multiple things